### PR TITLE
Propr update to include adjacency 

### PR DESCRIPTION
**Pull request by caraiz2001**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/propr/propr/templates/propr.R [hunk 1]
```diff
@@ -96,6 +96,43 @@ set_reference <- function(ivar, mat){
...
+    return(cutoff)
+}
+
+#' Convert a proportionality matrix to an adjacency matrix based on a threshold.
+#'
+#' @param matrix proportionality matrix. Can be extracted from propr object with getMatrix().
+#' @param cutoff Significant proportionality value extracted from valCutoff function.
+#'
+#' @return Adjacency matrix. Gene pairs with a proportionality value higher than the threshold will have 1, otherwise 0.
+convert_to_adjacency <- function(matrix, cutoff) {
```
Comment (suzannejin) [inline modules/nf-core/propr/propr/templates/propr.R:131]: some metrics like 'rho' and 'pcor' they are better with high values. But other metrics like 'phi' or 'phs' they are the contrary.
You should consider this when converting to adjacency matrix.

File: modules/nf-core/propr/propr/tests/main.nf.test [hunk 2]
```diff
@@ -159,5 +160,31 @@ nextflow_process {
...
+
+        tag "adjacency"
+        config "./adjacency.config"
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test' ],
+                    //file(params.test_data['mus_musculus']['genome']['rnaseq_matrix'], checkIfExists: true)
```
Comment (suzannejin) [inline modules/nf-core/propr/propr/tests/main.nf.test:173]: you can remove line 173 if you want

File: modules/nf-core/propr/propr/templates/propr.R (hunk 1 above)
Comment (caraiz2001) [reply modules/nf-core/propr/propr/templates/propr.R]: I have modified the convert_to_adjacency and valCutoff function to include this

File: modules/nf-core/propr/propr/templates/propr.R [hunk 3]
```diff
@@ -96,6 +96,79 @@ set_reference <- function(ivar, mat){
...
+#' Gene pairs with a proportionality value higher than the extracted cutoff will be considered significantly proportional.
+#'
+#' @param object propr object. Output from propr function. updateCutoffs function should be applied to the object previous to valCutoff.
+#' @param fdrVal FDR value to extract the cutoff for. Per default 0.05
+#' @param metric Metric used to calculate the proportionality values. Options are 'cor', 'rho', 'phi', 'phs', 'vlr', 'pcor', 'pcor.shrink', 'pcor.bshrink'
+#'
+#' @return cutoff value. Proportionality values higher than this cutoff are considered significant.
+valCutoff  <- function(object, metric, fdrVal = 0.05){
+    fdr_df <- object@fdr
+    print(fdr_df)
```
Comment (suzannejin) [inline modules/nf-core/propr/propr/templates/propr.R:126]: do you want to print it to log?

File: modules/nf-core/propr/propr/templates/propr.R [hunk 4]
```diff
@@ -96,6 +96,79 @@ set_reference <- function(ivar, mat){
...
+        warning("Please run updateCutoff on propr first")
+    }else{
+        fdr_vals <- fdr_df\$FDR
+        if (any(is.na(fdr_vals))){
+            stop("FDR not defined. This metric is not appropiate for the given dataset")
+        }
+        threshold <- any(fdr_vals <= fdrVal)
+        if (metric %in% metric_up){
+            if (threshold){
+                fdr_threshold <- fdr_vals[which.max(fdr_vals <= fdrVal)]
```
Comment (suzannejin) [inline modules/nf-core/propr/propr/templates/propr.R:138]: `fdr_vals <= fdrVal` returns a vector of boolean values no?
Does `which.max` or `which.min` work well with boolean values?

**Review by suzannejin — APPROVED** [review]

**Pull request merged**